tests/ontopy_tests/test_graph2.py

```diff
@@ -1,14 +1,43 @@
 from typing import TYPE_CHECKING
+import owlready2
+import pytest
+from ontopy.graph import OntoGraph
 
 if TYPE_CHECKING:
     from pathlib import Path
 
     from ontopy.ontology import Ontology
 
 
-def test_graph2(emmo: "Ontology", tmpdir: "Path") -> None:
-    from ontopy.graph import OntoGraph
+@pytest.mark.filterwarnings(
+    "ignore:Style not defined for relation hasSpecialRelation."
+)  # currently pytest is set to accept warnings, but this might change in the future
```

Comment:
Is it important for the test the warning is ignored?
Otherwise, I don't think it's relevant to ignore here. Rather one can either:

- Move the ignore statement to the general pytest configuration (`pyproject.toml`):

  ```ini
  [tool.pytest.ini_options]
  filterwarnings = [
      # Some comment here on why this is ignored.
      "ignore:Style not defined for relation hasSpecialRelation.",
  ]
  ```

- Remove the filterwarning ignore statement altogether.

Reply:
I want to eventually add that the pytest fails upong generation of warnings. Sometimes the warnings are desired, therefore this filterwarningstatement. I am adding explicitly an object property that does not have a default style, and I want this to recognized. I guess I could also use pytest.warns(), but I think this is enough.

Reply:
Filterwarnings _ignores_ warnings that occur, while `pytest.warns()` checks warnings _do_ occur. So, as I read the test you're _ignoring_ a specific warning in the output only. I'm pretty sure the warning is still emitted, just ignored in the printouts. If you want to test a specific warning is emitted, you should use `pytest.warns()`.

Reply:
Yes, in this specific example it ignores a very specific warning. If we at some point add (which I intend to) pytest failing on warnings, then it will fail if any other warning is emitted.

Reply:
Anyhow, changed it to testing for that exact warning now, so hope it is OK.